### ISSUE TITLE
Jumper settings missing

### ISSUE THREAD

**Issue by rclarke250** [makerbase-mks/MKS-Robin-Pro#2]
How to configure for MKS TMC2209 in UART mode. The Robin Nano has a nice WIKI, where is the WIKI for the PRO?

**Comment by dchandelier** [comment]
i have the same ask.
Do you receved some update?

**Comment by rclarke250** [comment]
Nope. I did figure out uart. 3rd jumper down left and center pins closed.
And make sure in platformIO for the Robin_pro that TMCstepper library is
not listed in ignore_lib. Also there is an issue in the .pins file if using
a lvgl touch screen. If needed, let me know. I can send it to you. I just
got my touch working.

**Comment by poisson21** [comment]
Hello, gladly take your solution like others, i can't seem to make uart work with tmc 2209

**Comment by rclarke250** [comment]
I can't get it to work consistent, one driver will act normal, and next
reboot, it will switch to another. Marlin did work on the pins file, so it
no longer gives an error and refuse to load the TMC softwareserialm library.

**Comment by tnap1979** [comment]
how are you even able to flash the firmware to this shit board??
ive tried everything and only thing that works is the OEM pre-compiled shit firmware from jgmaker...
i want to use latest marlin and add bltouch and tmc-2209 to my board, but the POS keeps hanging at 99%
and results in a display that just says " boot" on every power up...
i heard possible that it may be bootloader locked ?? or require encrypted firmware ??
i mean why else does the OEM pre-compiled .bin work, but every flavour of marlin i try hangs and fails...
my vscode and platformio stuff is all fully updated, as well as latest windows 10 and drivers

**Comment by rclarke250** [comment]
a lot to unpack here, I no longer have this board in my machine, but UART mode has a write up about changes needed to the pins file on the jgmaker facebook group. As to the screen, it also needs a change to the pins file, but I worked with marlin and that should be fixed. in the folder with the firmware.bin, that needs to be renamed to robinpro35.bin you will find an assets folder, this is the screen icons and touch settings, drag and drop on to the sd card with the firmware. there is no eeprom on the touch board, it uses space on the main board for this.

**Comment by shhht** [comment]
@rclarke250 I'm trying to get uart working, but i don't know what you meant with "center pins closed" and I'm not sure if i understand the jumper position (is it from top to bottom the 3rd and on the green pins?), would you mind clarify your explanation? Thanks in advance.

**Comment by rclarke250** [comment]
here is a write up someone did for the Artist-D.

**Comment by shhht** [comment]
@rclarke250 Thank you for responding. One problem. I can't find a link, did you forget to add a link?

**Comment by rclarke250** [comment]
https://www.dropbox.com/s/52t2sk1fy1zmx5i/2021-06-01%20UART%20Artist-D-1_230310_105649.pdf?dl=0
sorry, replied through email. Forgot it wouldn't just share the pdf.

**Comment by shhht** [comment]
It worked, thank you for helping me.
